test/tstQueryTreeDegenerate.cpp

```diff
@@ -137,6 +155,33 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(single_leaf_tree, TreeTypeTraits,
                              {{{5., 5., 5.}}, 2.},
                          }),
                          make_reference_solution<int>({0, 0}, {0, 1, 2, 2}));
+}
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(single_leaf_tree_nearest_predicate,
+                              TreeTypeTraits, TreeTypeTraitsList)
+{
+  using Tree = typename TreeTypeTraits::type;
+  using ExecutionSpace = typename TreeTypeTraits::execution_space;
+  using DeviceType = typename TreeTypeTraits::device_type;
+
+  // tree has a single leaf (unit box)
+  auto const tree =
+      make<Tree>(ExecutionSpace{}, {
+                                       {{{0., 0., 0.}}, {{1., 1., 1.}}},
+                                   });
+
+  BOOST_TEST(!tree.empty());
+  BOOST_TEST(tree.size() == 1);
+  BOOST_TEST(
+      ArborX::Details::equals(tree.bounds(), {{{0., 0., 0.}}, {{1., 1., 1.}}}));
```

Comment:
Have you considered delegating empty/one leaf/two leaves tree construction into a separate function in the file to be called by both tests in order to reduce code duplication?